### PR TITLE
kernel: enable CPU bandwidth provisioning

### LINKED ISSUE #251: failed to set CPU period and quota
see https://github.com/clearcontainers/agent/issues/195

### PR DESCRIPTION
**Pull request by devimc**
this patch is to enable CPU period and quota in cgroups
* CONFIG_CFS_BANDWIDTH
    
Fixes #251
Fixes https://github.com/clearcontainers/agent/issues/195

Signed-off-by: Julio Montes <julio.montes@intel.com>

### CODE REVIEW

**Comment by jcvenegas** [comment]
@devimc  this is for cgruops support inside the VM ?  Is this a mandatory kernel configuration?, if it is the case you may want to document somewhere that this will be used by the agent(?), if case is not mandatory  document the expected behavior of not being enabled.

**Comment by jcvenegas** [comment]
Travis is not happy, it wants a issue and a `Fixes`

**Comment by jodh-intel** [comment]
Once you've added a `Fixes #XXX`,

lgtm

/cc @grahamwhaley.

File: kernel/kernel-config-4.9.x [hunk 1]
```diff
@@ -157,12 +157,13 @@ CONFIG_MEMCG=y
...
 CONFIG_FAIR_GROUP_SCHED=y
-# CONFIG_CFS_BANDWIDTH is not set
+CONFIG_CFS_BANDWIDTH=y
 # CONFIG_RT_GROUP_SCHED is not set
 # CONFIG_CGROUP_PIDS is not set
 CONFIG_CGROUP_FREEZER=y
 # CONFIG_CGROUP_HUGETLB is not set
-# CONFIG_CPUSETS is not set
+CONFIG_CPUSETS=y
+CONFIG_PROC_PID_CPUSET=y
```
Comment (grahamwhaley) [inline kernel/kernel-config-4.9.x]: Just checking - do we need/use this PID_CPUSET fs - it appears to be marked 'legacy' in the Kconfig, and I think it is superseded by the cgroups fs:
https://github.com/clearcontainers/linux/blob/master/init/Kconfig#L821
https://github.com/clearcontainers/linux/blob/master/Documentation/cgroup-v1/cpusets.txt#L769

Comment (grahamwhaley) [reply kernel/kernel-config-4.9.x]: I can see why we need this so the agent can apply the cpu resource management for the containers within a pod for instance. Just checking - I expect we need this in the kata repos, but are we applying it here so we can develop the agent code in the meantime before switching over to kata?

Comment (devimc) [reply kernel/kernel-config-4.9.x]: @grahamwhaley ahh you are right, lemme remove PID_CPUSET

**Comment by devimc** [comment]
@jcvenegas yes, we need it to apply cgroups inside the POD

@jodh-intel @jcvenegas  @grahamwhaley changes applied, thanks

**Comment by jodh-intel** [comment]
lgtm

**Review by jcvenegas — CHANGES_REQUESTED** [review]
Hi can you increase the release in the file in https://github.com/clearcontainers/packaging/blob/master/kernel/release. This will help to create a new kernel and publish it for our ci usage. See https://github.com/clearcontainers/packaging/tree/master/kernel#send-pr-of-new-kernel-release.

**Comment by devimc** [comment]
@jcvenegas done

**Review by jcvenegas — APPROVED** [review]

**Comment by jcvenegas** [comment]
new kernel with PR changes: 
https://github.com/clearcontainers/linux/releases/download/v4.9.60-84.container/v4.9.60-84.container-binaries.tar.gz

**Pull request merged**